iquip/apps/builder.py

```diff
@@ -574,6 +582,110 @@ def scanArguments(self) -> Dict[str, Any]:
+        self.centerSpinBox = QDoubleSpinBox(self)
+        self.applyProperties(self.centerSpinBox)
+        self.centerSpinBox.setValue(state["center"] / self.scale)
+        self.spanSpinBox = QDoubleSpinBox(self)
+        self.applyProperties(self.spanSpinBox)
+        self.spanSpinBox.setValue(state["span"] / self.scale)
+        self.stepSpinBox = QDoubleSpinBox(self)
+        self.applyProperties(self.stepSpinBox)
+        self.stepSpinBox.setValue(state["step"] / self.scale)
+        self.randomizeCheckBox = QCheckBox("Randomize", self)
+        self.randomizeCheckBox.setChecked(state["randomize"])
+        # layout
+        self.layout.addWidget(QLabel("center:", self), 0, 0)
+        self.layout.addWidget(self.centerSpinBox, 0, 1)
+        self.layout.addWidget(QLabel("span:", self), 1, 0)
+        self.layout.addWidget(self.spanSpinBox, 1, 1)
+        self.layout.addWidget(QLabel("step:", self), 2, 0)
+        self.layout.addWidget(self.stepSpinBox, 2, 1)
+        self.layout.addWidget(self.randomizeCheckBox, 3, 1)
+
+    def scanArguments(self) -> Dict[str, Any]:
+        """Overridden.
+        
+        Returns the arguments of the center scan.
+        """
+        return {
+            "ty": "CenterScan",
+            "center": self.centerSpinBox.value(),
+            "step": self.stepSpinBox.value(),
+            "span": self.spanSpinBox.value(),
+            "randomize": self.randomizeCheckBox.isChecked(),
+            "seed": None
+        }
+
+
+class _ExplicitScan(_BaseScan):
+    """Widget for explicit scan in _ScanEntry.
+
+    Attributes:
+        centerSpinBox: QDoubleSpinBox for center argument inside state.
+        spanSpinBox: QDoubleSpinBox for span argument inside state.
+        stepSpinBox: QSpinBox for step argument inside state.
+        randomizeCheckBox: QCheckBox for randomize argument inside state.
+    """
+    def __init__(
+        self,
+        procdesc: Dict[str, Any],
+        state: Dict[str, Any],
+        parent: Optional[QWidget] = None
+        ):
+        """Extended.
+
+        Args:
+            state: A key and its value is:
+              sequence: The sequnce that describes ExplicitScan sequence.
+        """
+        super().__init__(procdesc, parent=parent)
+        self.sequenceEdit = QLineEdit(self)
+        # layout
+        self.layout.addWidget(QLabel("sequence:", self), 0, 0)
+        self.layout.addWidget(self.sequenceEdit, 0, 1)
+        float_regexp = r"(([+-]?\d+(\.\d*)?|\.\d+)([eE][+-]?\d+)?)"
+        regexp = "(float)?( +float)* *".replace("float", float_regexp)
+        self.sequenceEdit.setValidator(QRegExpValidator(QRegExp(regexp)))
+        self.sequenceEdit.setText(" ".join([str(x) for x in state["sequence"]]))
+
+    def scanArguments(self) -> Dict[str, Any]:
+        """Overridden.
+        
+        Returns the arguments of the explicit scan.
+        """
+        text = self.sequenceEdit.text()
+        explicitText = [float(x) for x in text.split()]
```

Comment:
This is not a string, but a list of float numbers, so the variable name looks invalid.
How about replacing `text` with `sequenceText` and `explicitText` with `sequence`?